src/lamp_py/bus_performance_manager/vehicle_events.py

```diff
@@ -0,0 +1,166 @@
+from datetime import timedelta, date
+from typing import Optional
+import re
+
+import polars as pl
+
+from lamp_py.aws.s3 import (
+    file_list_from_s3_with_details,
+    get_datetime_from_partition_path,
+    get_last_modified_object,
+)
+
+from lamp_py.runtime_utils.remote_files import RemoteFileLocations
+
+
+def get_new_event_files() -> pl.DataFrame:
```

Comment:
i replaced how this file works, using polars. it will generate a df where each record is a service date and a two lists (gtfs_rt and tm) of files associated with that service date. on dev it takes just over a second.